tests/components/smlight/test_init.py

```diff
@@ -55,19 +55,36 @@ async def test_async_setup_auth_failed(
     assert entry.state is ConfigEntryState.NOT_LOADED
 
 
+async def test_async_setup_missing_credentials(
+    hass: HomeAssistant,
+    mock_config_entry_host: MockConfigEntry,
+    mock_smlight_client: MagicMock,
+) -> None:
+    """Test we trigger reauth when credentials are missing."""
+    mock_smlight_client.check_auth_needed.return_value = True
+
+    with patch(
+        "homeassistant.components.smlight.config_flow.SmlightConfigFlow.async_step_reauth"
+    ) as mock_async_step_reauth:
+        await setup_integration(hass, mock_config_entry_host)
+        mock_async_step_reauth.assert_called_once()
```

Comment:
We should instead check if there is a flow in progress after this

Reply:
Ok, have changed it to check flow progress instead.